astropy/convolution/src/convolve.c

```diff
@@ -338,15 +379,22 @@ FORCE_INLINE void convolve2d_boundary_none(DTYPE * const result,
                         top += val * ker;
                 }}
             }}
+
+            if (padded) { // compile time constant
+              result_index = i_minus_wkx * ny_minus_2wky + j_minus_wky;
```

Comment:
@astrofrog Sorry I missed this before. Now that there is only a single C func, the stride (``ny_minus_2wky``) should not be implicit. The shape of the results array should be passed in.

This also adds to my point about the semantics of ``padded``. I think the only argument to make for not passing in the shape, i.e. that the implicit use above is correct, would be to state that ``padded`` referred only to the results array.  However, this would also require the inverse of the condition since, technically, the ``boundary=None`` case is the one with the "padded" results array.

Reply:
Same goes for the 3D case.